syntax = "proto3";

package pm;

service PortMapper {
  rpc register_service (RegisterServiceRequest) returns (RegisterServiceReply) {}
  rpc unregister_service (UnregisterServiceRequest) returns (UnregisterServiceReply) {}
  rpc get_service_info (GetServiceInfoRequest) returns (GetServiceInfoReply) {}
  rpc terminate (TerminateRequest) returns (TerminateReply) {}
}

message RegisterServiceRequest {
  string service_name = 1;
  double value = 2;
}

message RegisterServiceReply {
  int32 port = 1;
}

message UnregisterServiceRequest {
  int32 port = 1;
}

message UnregisterServiceReply {
  int32 success = 1;
}

message GetServiceInfoRequest{
  string service_name = 1;
}

message GetServiceInfoReply {
  int32 port = 1;
  double value = 2;
}

message TerminateRequest{
}

message TerminateReply {
  int32 num_services = 1;
}